### PR TITLE
make: use gotestsum for tests

### PR DESCRIPTION
**Pull request by stevekuznetsov**
This utility brings a lot of user-experience benefits to testing, as
well as emitting nice jUnit XML files that publish the output of our
tests for consumption by structured CI displays.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -125,19 +125,23 @@ $(OPENSHIFT_GOIMPORTS):
...
 test-e2e: WHAT ?= ./test/e2e...
 test-e2e: build-all
-	NO_GORUN=1 go test -race -count $(COUNT) -p $(E2E_PARALLELISM) -parallel $(E2E_PARALLELISM) $(WHAT) $(TEST_ARGS)
+	NO_GORUN=1 gotestsum $(GOTESTSUM_ARGS) -- -race -count $(COUNT) -p $(E2E_PARALLELISM) -parallel $(E2E_PARALLELISM) $(WHAT) $(TEST_ARGS)
 
 .PHONY: test
 test: WHAT ?= ./...
 test:
-	go test -race -count $(COUNT) -coverprofile=coverage.txt -covermode=atomic $(TEST_ARGS) $$(go list "$(WHAT)" | grep -v ./test/e2e/)
+	 gotestsum $(GOTESTSUM_ARGS) -- -race -count $(COUNT) -coverprofile=coverage.txt -covermode=atomic $(TEST_ARGS) $$(go list "$(WHAT)" | grep -v ./test/e2e/)
```
Comment (ncdc) [inline Makefile:144]: Indentation is a little off?

**Review by ncdc — CHANGES_REQUESTED** [review]
At least for local development (if not in CI too), could you please install gotestsum via the `$(GO_INSTALL)` convention we are using elsewhere in the Makefile?

**Comment by ncdc** [comment]
Note, this makes it so `make test TEST_ARGS=-v` (verbose test output) doesn't work any more. You instead have to do `make test GOTESTSUM_ARGS='-f standard-verbose'`. I'm ok with that, but it was unexpected. Anyone feel strongly against this? @sttts

**Review by ncdc — APPROVED** [review]

**Comment by stevekuznetsov** [comment]
Part of #324

**Pull request merged**